pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/util/autocomplete/AstPackageExplorer.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.util.fxdesigner.util.autocomplete;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Modifier;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import net.sourceforge.pmd.lang.Language;
+
+
+/**
+ * Finds XPath node names by looking into the classpath
+ * directory corresponding to the AST of a language. This
+ * is ok for Java, Apex, etc. but not e.g. for XML.
+ */
+class AstPackageExplorer implements NodeNameFinder {
+    private List<String> availableNodeNames = new ArrayList<>();
```

Comment:
this initialization is redundant given that it's initialized in the constructor. Also, the list could be made final by removing this initialization.

Reply:
That's right, will do